### ISSUE TITLE
BinaryEncoder assigns labels starting from 1

### ISSUE THREAD

**Issue by karolsputo** [scikit-learn-contrib/category_encoders#195]
I'm trying to encode some categorical variables and I found that I'm getting too much dimensions due to BinaryEncoder starting labeling from 1 instead of 0.

I'm working on the same mushroom data as in the examples (described in this blog post in detail http://www.willmcginnis.com/2015/11/29/beyond-one-hot-an-exploration-of-categorical-variables/).

In there Will is using a simple BinaryEncoder without any additional parameters and he's getting 44 dimensions.
```
encoder().fit_transform(X, y)
```

I'm using the BinaryEncoder with drop_invariant option and getting 56 dimensions.
```
encoder_drop = ce.BinaryEncoder(cols = columnsToEncode, return_df=True, drop_invariant=True)
encoder_drop.fit(X)
encodedpreProcessed_drop = encoder_drop.transform(X)
```

After counting each columns I found one that there are no records with 0 bit representations. So every columns is split into more columns than necessary (+1).
```
encodedpreProcessed_drop.loc[(encodedpreProcessed_drop['cap-shape_1']==0) & (encodedpreProcessed_drop['cap-shape_2']==0) & (encodedpreProcessed_drop['cap-shape_3']==0)]
```
<img width="295" alt="Screen Shot 2019-06-06 at 00 28 25" src="https://user-images.githubusercontent.com/35783431/58994793-0fe4d080-87f2-11e9-8604-ee74719036bf.png">

Am I doing something wrong or is this now an intentional representation? Seems like I'm getting unnecessary columns thanks to this.

**Comment by janmotl** [comment]
1. I guess we have to add seed to cross-validation (maybe with `np.random.seed(value)`?) in order to get reproducible results.
2. I guess that with `handle_missing='error'` and `handle_unknown='error'` you should get fewer features (some combinations of the values can be reserved for handling missing and previously unseen values in the scoring set).

Does the second paragraph solve the issue? If yes, how would you recommend to change the documentation to communicate this behaviour?

**Comment by kerrickstaley** [comment]
The value 0 (i.e. all the columns in the output are 0) is reserved for representing missing and unknown values in the input. So I don't think there is a real bug here, just a misunderstanding of the behavior.

Separately, `BinaryEncoder` usually uses one more bit than needed in general. That issue is filed [here](https://github.com/scikit-learn-contrib/category_encoders/issues/264) and I have a fix [here](https://github.com/scikit-learn-contrib/category_encoders/pull/296).

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — this one looks like a duplicate of #264.

The off-by-one observation that BinaryEncoder uses one more bit than strictly necessary is tracked in #264, with a fix proposed in [PR #296](https://github.com/scikit-learn-contrib/category_encoders/pull/296). The all-zero row in the user's screenshot is the by-design reservation for the `handle_missing`/`handle_unknown` sentinel ([`@kerrickstaley pointed this out in 2021`](https://github.com/scikit-learn-contrib/category_encoders/issues/195#issuecomment-817402405)).

If you agree, would you mind closing in favour of #264 (and adding a 👍 there if you want to signal demand)? Happy to reopen if the symptoms diverge.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the cross-link was verified manually against the current `master` branch.)

**Comment by wdm0006** [comment]
Starting at 1 is intentional: the all-zero row is reserved as the missing/unknown sentinel, consistent with how the rest of the library treats unknown values, and it's what keeps `inverse_transform` round-trips unambiguous. Changing it would silently alter fitted mappings for existing users and interact badly with `drop_invariant`. Closing as by design — reopen if you hit a concrete downstream problem.
